go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -0,0 +1,290 @@
+package sampledecoder
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/observability-lib/ovsdb"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/observability"
+)
+
+type SampleDecoder struct {
+	nbClient          client.Client
+	ovsdbClient       client.Client
+	cleanupCollectors []int
+}
+
+type nbConfig struct {
+	address string
+	scheme  string
+}
+
+type Cookie struct {
+	ObsDomainID uint32
+	ObsPointID  uint32
+}
+
+const CookieSize = 8
+const bridgeName = "br-int"
+
+var SampleEndian = getEndian()
+
+func getEndian() binary.ByteOrder {
+	// Use network bite order
+	return binary.BigEndian
+}
+
+func getLocalNBClient(ctx context.Context) (client.Client, error) {
+	config := nbConfig{
+		address: "unix:/var/run/ovn/ovnnb_db.sock",
+		scheme:  "unix",
+	}
+	libovsdbOvnNBClient, err1 := NewNBClientWithConfig(ctx, config)
+	if err1 == nil {
+		return libovsdbOvnNBClient, nil
+	}
+	config = nbConfig{
+		address: "unix:/var/run/ovn-ic/ovnnb_db.sock",
+		scheme:  "unix",
+	}
+	var err2 error
+	libovsdbOvnNBClient, err2 = NewNBClientWithConfig(ctx, config)
+	if err2 != nil {
+		return nil, fmt.Errorf("error creating libovsdb client: [ with /var/run/ovn/ovnnb_db.sock: %w, with /var/run/ovn-ic/ovnnb_db.sock: %w ]", err1, err2)
```

Comment:
Nit: maybe don't hardcode the paths, we have the last one in the `config` variable.  We could, I guess, use two `config` variables.

Also, this kind of gives the impression that we support non-IC, while in reality we don't. :)